File: 프로그래머스/0/181900. 글자 지우기/글자 지우기.py
def solution(my_string, indices):
    # indices를 set으로 변환하여 조회 속도 최적화 (O(1))
    remove_set = set(indices)
    
    answer = []
    for idx, ch in enumerate(my_string):
        # 지울 인덱스 목록에 포함되어 있지 않은 문자만 남김
        if idx not in remove_set:
            answer.append(ch)
            
    # 리스트의 문자들을 하나의 문자열로 합쳐서 반환
    return "".join(answer)

File: 프로그래머스/0/181900. 글자 지우기/글자 지우기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string my_string, vector<int> indices) {
    string answer = "";
    
    // 지울 인덱스를 표시할 bool 배열 생성 (기본값 false)
    vector<bool> is_removed(my_string.size(), false);
    
    // 지워야 하는 인덱스는 true로 변경
    for (int idx : indices) {
        is_removed[idx] = true;
    }
    
    // false인 인덱스의 글자만 answer에 추가
    for (int i = 0; i < my_string.size(); i++) {
        if (!is_removed[i]) {
            answer += my_string[i];
        }
    }
    
    return answer;
}